docs/how-to-add-new-crd-and-controller.md

```diff
@@ -0,0 +1,151 @@
+	config/rbac/operator_demo_viewer_role.yaml
+	config/samples/operator_v1alpha1_demo.yaml
+	controllers/operator/demo_controller.go
+	controllers/operator/suite_test.go
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+Next, we need to focus on the file `apis/operator/v1alpha1/demo_types.go` which defines your CRD, and the file `controllers/operator/configuration_controller.go` which defines the Controller. The others files are some configurations generated by the [kubebuilder markers](https://book.kubebuilder.io/reference/markers.html). Here are some references:
+
+* [Kubebuilder project demo](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3), in which you can understand the overall architecture.
+
+* [How to add new-api](https://book.kubebuilder.io/cronjob-tutorial/new-api.html), which you can find more details for `oapserverconfig_types.go`.
+
+* [Controller-overview](https://book.kubebuilder.io/cronjob-tutorial/controller-overview.html), where you can find more details about `oapserverconfig_controller.go`.
+
+At last, the `suite_test.go` can be deleted as we don't need it.
+
+
+
+### 3. Create webhook
+
+If you want to validate the field of your CRD, you need to create the webhook as follows.
+
+```sh
+kubebuilder create webhook --group operator --version v1alpha1 --kind Demo --defaulting --programmatic-validation
+```
+
+Below is the newly generated file.
+
+```sh
+$ git status
+On branch master
+Your branch is ahead of 'origin/master' by 1 commit.
+  (use "git push" to publish your local commits)
+
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+	modified:   PROJECT
+	modified:   config/webhook/manifests.yaml
+	modified:   main.go
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	apis/operator/v1alpha1/demo_webhook.go
+	apis/operator/v1alpha1/webhook_suite_test.go
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+You can get more details through [webhook-overview](https://book.kubebuilder.io/reference/webhook-overview.html), and you also need to delete the file `webhook_suite_test.go`.
+
+
+
+### 4. Create the template
+
+Generally, a CRD will be associated with a series of resources, such as [workload](https://kubernetes.io/docs/concepts/workloads/), [rbac](https://kubernetes.io/docs/reference/access-authn-authz/rbac/), [service](https://kubernetes.io/docs/concepts/services-networking/service/), etc. All template files are stored in the `./operator/pkg/operator/manifests`, you could create the directory there such as `demo`, and refer to the CRD templates in other folders to define your own template. What's more, we support defining your own template rendering function by setting up the [TmplFunc](https://github.com/apache/skywalking-swck/blob/master/operator/pkg/kubernetes/apply.go#L49).
+
```

Comment:
Also, after adding the template, we have to change the comments here https://github.com/apache/skywalking-swck/blob/bf4d1346a9869f67187b9b9202bf14d190728c56/operator/pkg/operator/manifests/repo.go#L31 and add `demo`.